src/YunInternet/Libvirt/XMLImplement/SimpleXMLImplement.php

```diff
@@ -72,8 +72,13 @@ public function getChildren($name, $filter = null, $wrapper = null): array
             };
         }
 
+        return $this->GetCollection($name, $filter, $wrapper, $nested, $this->simpleXMLElement);
+    }
+
+    private function GetCollection(string $name, $filter, object $wrapper, bool $nested, $xml)
```

Comment:
Please use lowerCamelCase style method name. Only begin with upper case when it's a abbreviation, like IOTune, which is the abbreviation of "Input Output Tune".
You don't have to invoke this function recursively, there is an method SimpleXMLElement::xpath(), which can simplify your code, like this:
```
...
$path = ".//" . $name;
foreach ($this->simpleXMLElement->xpath($path) as $child) {
    ...
}
...
```
$this->simpleXMLElement->{$name} can be replaced with XPath too: 
```
$path = "./" . $name
```

Reply:
Thank for your suggestion